tc/aten/aten_compiler_new_api-inl.h

```diff
@@ -0,0 +1,145 @@
+  auto parsedTcs = tc::detail::parse(tc);
+  CHECK_EQ(1, parsedTcs.count(entryPoint))
+      << "attempting to access undefined function " << entryPoint;
+
+  std::vector<at::Tensor> outputs;
+  auto inputDLTensors = toDLConstTensors(inputs);
+  auto outTensorInfo = detail::inferOutputTensorInfo(
+      parsedTcs.at(entryPoint), extractRawPtrs(inputDLTensors));
+  if (outTensorInfo.size() == 0) {
+    return outputs;
+  }
+  CHECK_LT(0, inputs.size())
+      << "Need >= 1 input tensors to determine backend and prepare ATen outputs";
+  auto atenBackend = inputs[0].type().backend();
+  for (size_t i = 0; i < outTensorInfo.size(); ++i) {
+    auto info = outTensorInfo[i];
+    auto stype = at::toScalarType(info.dtype);
+    outputs.push_back(
+        at::getType(atenBackend, stype).tensor(at::IntList(info.shape)));
+    // TODO: we just malloc'ed I guess we can pay an inplace memset
+    outputs.back().zero_();
+  }
+  return outputs;
+}
+
+template <typename Backend>
+std::unique_ptr<typename Backend::ExecutorType> compile(
+    const std::string& tc,
+    const std::string& entryPoint,
+    const std::vector<at::Tensor>& inputs,
+    const typename Backend::MappingOptionsType& options) {
+  auto inputDLTensors = toDLConstTensors(inputs);
+  return tc::compile<Backend>(
+      tc, entryPoint, extractRawPtrs(inputDLTensors), options);
+}
+
+template <typename Executor>
+void run(
+    const Executor& executor,
+    const std::vector<at::Tensor>& inputs,
+    std::vector<at::Tensor>& outputs) {
+  auto inputDLTensors = toDLConstTensors(inputs);
+  auto outputDLTensors = makeDLTensors(outputs);
+  return executor.run(
+      extractRawPtrs(inputDLTensors), extractRawPtrs(outputDLTensors));
+}
+
+template <typename Executor>
+ProfilingInfo profile(
+    const Executor& executor,
+    const std::vector<at::Tensor>& inputs,
+    std::vector<at::Tensor>& outputs) {
+  auto start = std::chrono::system_clock::now();
+  auto inputDLTensors = toDLConstTensors(inputs);
+  auto outputDLTensors = makeDLTensors(outputs);
+  ProfilingInfo pi(executor.profile(
+      extractRawPtrs(inputDLTensors), extractRawPtrs(outputDLTensors)));
+
+  // The total CPU overhead is the total time minus the (synchronized) kernel
+  // runtime
+  auto end = std::chrono::system_clock::now();
+  Duration cpuOverhead(end - start);
+  cpuOverhead = cpuOverhead - pi.kernelRuntime;
+  return ProfilingInfo{cpuOverhead, pi.kernelRuntime};
+}
+
+template <typename Executor>
+void uncheckedRun(
+    const Executor& executor,
+    const std::vector<at::Tensor>& inputs,
+    std::vector<at::Tensor>& outputs) {
+  CHECK_LT(0, outputs.size());
+
+  constexpr auto kReservedSize = 8;
+  std::vector<const void*> I(kReservedSize, nullptr);
+  std::vector<void*> O(kReservedSize, nullptr);
+  size_t i;
+  for (i = 0; i < inputs.size(); ++i) {
+    if (i < kReservedSize) {
+      I[i] = inputs[i].data_ptr();
+    } else {
+      I.push_back(inputs[i].data_ptr());
+    }
+  }
+  I.resize(i);
```

Comment:
Please use `I.resize(inputs.size())` instead.
If you are really worried about recomputing this value, then compute it up front and
store it in a variable that is also used in the for loop condition.